src/sssom/validators.py

```diff
@@ -1,51 +1,124 @@
+def print_linkml_report(report: ValidationReport, fail_on_error: bool = True):
+    """Print the error messages in the report. Optionally throw exception.
+
+    :param report: A LinkML validation report
+    :param fail_on_error: if true, the function will throw an ValidationError exception when there are errors
+    """
+    validation_errors = 0
+
+    if not report.results:
+        logging.info("The instance is valid!")
+    else:
+        for result in report.results:
+            validation_errors += 1
+            if result.severity == Severity.ERROR:
+                logging.error(result.message)
+
+    if fail_on_error and validation_errors:
+        raise ValidationError(f"You mapping set has {validation_errors} validation errors!")
+
+
+# TODO This should not be necessary: https://github.com/linkml/linkml/issues/2117,
+#  https://github.com/orgs/linkml/discussions/1975
+def _clean_dict(d):
+    """Recursively removes key-value pairs from a dictionary where the value is None, "null", or an empty string.
+
+    Args:
+    d (dict): The dictionary to clean.
+
+    Returns:
+    dict: A cleaned dictionary with unwanted values removed.
+    """
+    if not isinstance(d, dict):
+        return d
+
+    cleaned_dict = {}
+    for k, v in d.items():
+        if isinstance(v, dict):
+            # Recursively clean nested dictionary
+            cleaned_value = _clean_dict(v)
+            if cleaned_value:  # Add only if the nested dictionary is not empty
+                cleaned_dict[k] = cleaned_value
+        elif isinstance(v, list):
+            # Recursively clean list elements if they are dictionaries
+            cleaned_list = [_clean_dict(item) if isinstance(item, dict) else item for item in v]
+            # Filter out empty dictionaries from list
+            cleaned_list = [item for item in cleaned_list if item not in [None, "", "null", {}]]
+            if cleaned_list:  # Add only if the list is not empty
+                cleaned_dict[k] = cleaned_list
+        elif v not in [None, "", "null"]:
+            cleaned_dict[k] = v
+
+    return cleaned_dict
+
+
+def validate_json_schema(msdf: MappingSetDataFrame, fail_on_error: bool = True) -> None:
     """Validate JSON Schema using linkml's JsonSchemaDataValidator.
 
     :param msdf: MappingSetDataFrame to eb validated.
+    :param fail_on_error: if true, the function will throw an ValidationError exception when there are errors
     """
-    validator = ReferenceValidator(SchemaView(SCHEMA_YAML))
+    validator = Validator(
+        schema=SCHEMA_YAML,
+        validation_plugins=[JsonschemaValidationPlugin(closed=False), RecommendedSlotsPlugin()],
+    )
     mapping_set = to_mapping_set_document(msdf).mapping_set
-    validator.validate(mapping_set, MappingSet)
 
+    mapping_set_yaml = asdict(mapping_set)
+    mapping_set_yaml_cleaned = _clean_dict(mapping_set_yaml)
+
+    report = validator.validate(mapping_set_yaml_cleaned, "mapping set")
```

Comment:
@pkalita-lbl This is where https://github.com/linkml/linkml/issues/2164 is happening.. Maybe the three lines above that one are the problem?